### ISSUE TITLE
Signal collapse

### ISSUE THREAD

**Issue by initialAspiration** [YuhangLiu98/SDCNN#2]
Hello Author,  
I'm encountering an issue while training the model:  
- ​Input Data​​: Normalized to [0,1] range  
- ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.  

Could you please advise on how to resolve this issue?

**Comment by vie131313** [comment]
> Hello Author, I'm encountering an issue while training the model:
> 
> * ​Input Data​​: Normalized to [0,1] range
> * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> 
> Could you please advise on how to resolve this issue?

HI,can you run this normally?I tried, but it always stuck in the data load part,strange

**Comment by YuhangLiu98** [comment]
> Hello Author, I'm encountering an issue while training the model:
> 
> * ​Input Data​​: Normalized to [0,1] range
> * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> 
> Could you please advise on how to resolve this issue?

I haven't encountered this issue myself, but I have a few suggestions that might help:

1. Training Dataset Size: It's possible that the training dataset is too small. I recommend trying a larger dataset, such as the publicly available Mayo dataset, to see if that resolves the issue.
2. Test Data: I’ve also uploaded some of my test data, which you can use to assist in troubleshooting the problem.
3. Check prep.py:I suggest looking into the prep.py file, particularly how I perform normalization. Understanding the normalization process may provide some insights.

**Comment by YuhangLiu98** [comment]
> > Hello Author, I'm encountering an issue while training the model:
> > 
> > * ​Input Data​​: Normalized to [0,1] range
> > * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> > 
> > Could you please advise on how to resolve this issue?
> 
> HI,can you run this normally?I tried, but it always stuck in the data load part,strange

I have tested the code, and it runs normally. Could you please provide more details about the issue you're facing? You can try using the latest data I uploaded. Set it up as the training and testing sets, and it should run properly.

**Comment by vie131313** [comment]
> > > Hello Author, I'm encountering an issue while training the model:
> > > 
> > > * ​Input Data​​: Normalized to [0,1] range
> > > * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> > > 
> > > Could you please advise on how to resolve this issue?
> > 
> > 
> > HI,can you run this normally?I tried, but it always stuck in the data load part,strange
> 
> I have tested the code, and it runs normally. Could you please provide more details about the issue you're facing? You can try using the latest data I uploaded. Set it up as the training and testing sets, and it should run properly.

作者您好，我已经成功运行起来了您的代码,忘记删除这个评论了。除此之外作者我还有一些疑问。
1.我在mayo2016 的k30的1mm上的数据集进行训练，用L506作为测试数据，其他作为训练数据，每个epoch都测试一次。我发现训练很慢（训练集5410张图片，测试集526，一个epoch大概半个小时在3090i上），这是正常的吗？
2.而且测试的结果很快（几个epoch）就达到了最优值大概在26.93（后面一直在持续下降，似乎很快就会拟合，并不需要您论文中所说的200或300个epoch？请问这是因为数据集的原因吗？因为我看咱们用的数据集不同，您用的是mayo数据集自己选的10个case进行训练和测试，结果能达到接近33，希望得到您的建议，非常感谢！
3.初次之外我还有一个问题，比如psnr的结果是”32.95±0.97“，这个±0.97是跑多个seed的epoch得到结果计算的均值和标准差还是跑一次之后测试集上的均值和标准差呢，如果是seed，请问作者一般跑多少次比较合适呢？不对，我看到您的LDCT甚至都有均值和标准差是不是意味着这是测试集一次测试之后的测试集上的均值和标准差呢？因为按道理来说LDCT这个跑多次是不会变的。希望得到作者您的建议，非常感谢！！！

**Comment by YuhangLiu98** [comment]
> > > > Hello Author, I'm encountering an issue while training the model:
> > > > 
> > > > * ​Input Data​​: Normalized to [0,1] range
> > > > * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> > > > 
> > > > Could you please advise on how to resolve this issue?
> > > 
> > > 
> > > HI,can you run this normally?I tried, but it always stuck in the data load part,strange
> > 
> > 
> > I have tested the code, and it runs normally. Could you please provide more details about the issue you're facing? You can try using the latest data I uploaded. Set it up as the training and testing sets, and it should run properly.
> 
> 作者您好，我已经成功运行起来了您的代码,忘记删除这个评论了。除此之外作者我还有一些疑问。 1.我在mayo2016 的k30的1mm上的数据集进行训练，用L506作为测试数据，其他作为训练数据，每个epoch都测试一次。我发现训练很慢（训练集5410张图片，测试集526，一个epoch大概半个小时在3090i上），这是正常的吗？ 2.而且测试的结果很快（几个epoch）就达到了最优值大概在26.93（后面一直在持续下降，似乎很快就会拟合，并不需要您论文中所说的200或300个epoch？请问这是因为数据集的原因吗？因为我看咱们用的数据集不同，您用的是mayo数据集自己选的10个case进行训练和测试，结果能达到接近33，希望得到您的建议，非常感谢！ 3.初次之外我还有一个问题，比如psnr的结果是”32.95±0.97“，这个±0.97是跑多个seed的epoch得到结果计算的均值和标准差还是跑一次之后测试集上的均值和标准差呢，如果是seed，请问作者一般跑多少次比较合适呢？不对，我看到您的LDCT甚至都有均值和标准差是不是意味着这是测试集一次测试之后的测试集上的均值和标准差呢？因为按道理来说LDCT这个跑多次是不会变的。希望得到作者您的建议，非常感谢！！！

1.  对于训练速度，由于不知道3090i的算力以及你使用的CPU等情况，我无法判断是否正常。我可以给你分享我的情况，A6000平台上训练集1427对CT图像，一个epoch时间为3分半左右。
2. 建议尝试：训练集为L006-L064（共10个case），测试集为L004。
3. 正如你所说，0.97是一次测试之后的测试集PSNR指标的方差。

**Comment by vie131313** [comment]
> > > > > Hello Author, I'm encountering an issue while training the model:
> > > > > 
> > > > > * ​Input Data​​: Normalized to [0,1] range
> > > > > * ​Problem​: After passing through the `Genclean` network, the output values become extremely small (near-zero), causing all generated images to appear completely black. This prevents the neural network from learning meaningful features.
> > > > > 
> > > > > Could you please advise on how to resolve this issue?
> > > > 
> > > > 
> > > > HI,can you run this normally?I tried, but it always stuck in the data load part,strange
> > > 
> > > 
> > > I have tested the code, and it runs normally. Could you please provide more details about the issue you're facing? You can try using the latest data I uploaded. Set it up as the training and testing sets, and it should run properly.
> > 
> > 
> > 作者您好，我已经成功运行起来了您的代码,忘记删除这个评论了。除此之外作者我还有一些疑问。 1.我在mayo2016 的k30的1mm上的数据集进行训练，用L506作为测试数据，其他作为训练数据，每个epoch都测试一次。我发现训练很慢（训练集5410张图片，测试集526，一个epoch大概半个小时在3090i上），这是正常的吗？ 2.而且测试的结果很快（几个epoch）就达到了最优值大概在26.93（后面一直在持续下降，似乎很快就会拟合，并不需要您论文中所说的200或300个epoch？请问这是因为数据集的原因吗？因为我看咱们用的数据集不同，您用的是mayo数据集自己选的10个case进行训练和测试，结果能达到接近33，希望得到您的建议，非常感谢！ 3.初次之外我还有一个问题，比如psnr的结果是”32.95±0.97“，这个±0.97是跑多个seed的epoch得到结果计算的均值和标准差还是跑一次之后测试集上的均值和标准差呢，如果是seed，请问作者一般跑多少次比较合适呢？不对，我看到您的LDCT甚至都有均值和标准差是不是意味着这是测试集一次测试之后的测试集上的均值和标准差呢？因为按道理来说LDCT这个跑多次是不会变的。希望得到作者您的建议，非常感谢！！！
> 
> 1. 对于训练速度，由于不知道3090i的算力以及你使用的CPU等情况，我无法判断是否正常。我可以给你分享我的情况，A6000平台上训练集1427对CT图像，一个epoch时间为3分半左右。
> 2. 建议尝试：训练集为L006-L064（共10个case），测试集为L004。
> 3. 正如你所说，0.97是一次测试之后的测试集PSNR指标的方差。
好的，收到，谢谢您！！
